docusaurus/docs/iOS/uikit/controllers/channels.md

```diff
@@ -2,38 +2,108 @@
+
+The `query.filter` determines a set of conditions the channel should satisfy to match the query. It can contains conditions for **built-in** channel fields:
+```swift
+let filter1: Filter<ChannelListFilterScope> = .equal(.type, to: .messaging)
 ```
-client.channelListController(query: query, filter: { channel in
-    // Your filtering logic
+as well as conditions for **custom** fields:
+```swift
+let filter2: Filter<ChannelListFilterScope> = .equal("state", to: "LA")
+```
+Primitive filters can be **combined** using `and/or/nor` operators which allows to get a filter of any complexity:
+```swift
+let compoundFilter: Filter<ChannelListFilterScope> = .and([
+    filter1,
+    filter2
+])
+```
+
+The `query.sort` is an array of sorting options. Sorting option are applied based on their order in array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
+
+### 2. Create a controller
+
+The controller can be created using `ChatClient`:
+```swift
+let controller = ChatClient.shared.channelListController(query: query, filter: { channel in
+    // the filtering logic goes here.
     return true
 })
 ```
 
-It is expected that the logic on the `filter` block matches the one in `query.filter`.
+The `filter` closure is needed to dynamically link/unlink channels to the list during the controller's lifecycle to keep the list in sync with the remote:
+
+When `filter` returns `true`, the new/updated channel is linked/stays linked to the channel list
+
+When `filter` returns `false`, the new/updated channel is not linked/is unlinked from the channel list.
+
+:::note
+It is expected that the logic on the `filter` closure matches the one in `query.filter`.
+:::
+
+:::note
+Given the complex nature of `Filter`, having generic logic capable of using `query.filter` at runtime to determine if a channel meets the criteria is extremely expensive and complex. Instead, having a block that checks for the particular needs of the use case is much cheaper.
+:::
+
+### 3. Set the delegate
+
+An instance of the type conforming to `ChatChannelListControllerDelegate` protocol can be assigned as controller's delegate:
+```swift
+controller.delegate = delegate
+```
 
 :::note
-Given the nature of `Filter`, having generic logic capable of using `query.filter` at runtime to determine if a ChatChannel meets the criteria is extremely expensive and complex. Instead, having a block that only checks for the particular needs of the use case is much cheaper.
+An integrator should make sure to keep a strong reference to the `delegate` passed to the controller. Otherwise, the delegate object will get deallocated since the controller references it **weakly**.
 :::
 
+### 4. Synchronize controller
+
+The `synchronize` should be called on the controller to:
+- fetch the first page of channels matching the query
+- subscribe on events for those channels
+- setup the database observers
```

Comment:
The database observers are an implementation detail of the controller, and so the customers don't know, and shouldn't know about this. What about "start observing data changes", WDYT? 
```suggestion
The `synchronize` should be called on the controller to:
- fetch the first page of channels matching the query
- subscribe to events for those channels
- start observing data changes
```

Reply:
Fixed ✅